src/interactors/users/v1.ts

```diff
@@ -0,0 +1,16 @@
+import {User, UserFilters} from "../../types";
+import {getCachedUsers} from "../../services";
+import {UserRepository} from "../../repository/user-repository";
+
+export class InteractorUsersV1 {
+    async getUsers(filters: UserFilters): Promise<User[]> {
+        const cached = await getCachedUsers(filters);
+
+        if (!cached) {
+            const userRepository = new UserRepository();
```

Comment:
Maybe interactor should not create objects by itself, prefer to inject dependencies through constructor.